api/src/paths/dwc/spatial/download.ts

```diff
@@ -0,0 +1,105 @@
+import AdmZip from 'adm-zip';
+import { RequestHandler } from 'express';
+import { Operation } from 'express-openapi';
+import { Feature } from 'geojson';
+import { getAPIUserDBConnection, getDBConnection } from '../../../database/db';
+import { defaultErrorResponses } from '../../../openapi/schemas/http-responses';
+import { SpatialService } from '../../../services/spatial-service';
+import { getLogger } from '../../../utils/logger';
+
+const defaultLog = getLogger('paths/dwc/eml/download');
+
+export const GET: Operation = [downloadSpatialComponents()];
+
+GET.apiDoc = {
+  description: 'Archived search results for spatial components.',
+  tags: ['spatial', 'search', 'download'],
+  security: [
+    {
+      OptionalBearer: []
+    }
+  ],
+  parameters: [
+    {
+      in: 'query',
+      name: 'boundary',
+      required: true,
+      schema: {
+        type: 'string'
+      },
+      description: 'A stringified GeoJSON Feature. Will return results that intersect the feature.'
+    },
+    {
+      in: 'query',
+      name: 'type',
+      schema: {
+        type: 'array',
+        items: {
+          type: 'string'
+        },
+        nullable: true
+      },
+      description: 'An array of spatial component types to filter on. Will return results that match any of the types.'
+    },
+    {
+      in: 'query',
+      name: 'datasetID',
+      required: true,
+      schema: {
+        type: 'array',
+        items: {
+          type: 'string'
+        }
+      },
+      description: 'An array of dataset IDs. Will return results that belong to any of the dataset IDs.'
+    }
+  ],
+  responses: {
+    200: {
+      description: 'Archived spatial components response object.',
+      content: {
+        'application/zip': {
+          schema: {
+            type: 'string',
+            description: 'Buffer of the arcived search results.'
+          }
+        }
+      }
+    },
+    ...defaultErrorResponses
+  }
+};
+
+export function downloadSpatialComponents(): RequestHandler {
+  return async (req, res) => {
+    const criteria = {
+      type: (req.query.type as string[]) || [],
+      datasetID: req.query.datasetID as string[],
+      boundary: JSON.parse(req.query.boundary as string) as Feature
+    };
+
+    const connection = req['keycloak_token'] ? getDBConnection(req['keycloak_token']) : getAPIUserDBConnection();
+
+    try {
+      await connection.open();
+      const spatialService = new SpatialService(connection);
+      const response = await spatialService.findSpatialComponentsByCriteria(criteria);
+      await connection.commit();
```

Comment:
Shouldn't `connection.commit()` occur at the end of the function, just before `res.status(200).send(zipToSend.toString('hex'))`? I'm not entirely certain, although that is the convention for most of our other endpoints.

Reply:
moved `connection.commit()` to be consistent with the other endpoints

Reply:
We can commit as soon as we are done making any database calls, though that being said, it realistically probably doesn't matter too much as nothing heavy is happening after we finish calling the service in this case.